File: src/packagename/src/new_keyboard.cpp
// This program subscribes to turtle1/cmd_vel and republished on "turtle1/cmd_vel_reversed" topic with signs inverted

#include <ros/ros.h>
#include <geometry_msgs/Twist.h>
#include <barc/ECU.h>
#include <math.h>
#include <stdio.h>


ros::Publisher *pubPtr;
barc::ECU msgOut;

void commandVelocityReceived(const geometry_msgs::Twist& msgIn){

	msgOut.motor = round((msgOut.motor+msgIn.linear.x*0.01)*10000)/10000;
	/*if (msgOut.motor <0){
		msgOut.motor  = 0;
	}*/
	msgOut.servo = round((msgOut.servo+msgIn.angular.z*0.01)*10000)/10000;
	pubPtr->publish(msgOut);
}

int main(int argc, char **argv) {

	ros::init(argc,argv, "reverse_velocity_node");

	ros::NodeHandle nh;
	pubPtr = new ros::Publisher(nh.advertise<barc::ECU>("/ecu",1000));

	ros::Subscriber sub = nh.subscribe("turtle1/cmd_vel",1000,&commandVelocityReceived);

	ros::spin();

	delete pubPtr;

	}

File: src/packagename/src/new_keyboard.py
#!/usr/bin/env python

import rospy
import time
from geometry_msgs.msg import Twist
from barc.msg import ECU


# update
def callback_function(data):
    newECU.motor = newECU.motor + round(data.linear.x*0.01,5)       # input acceleration
    newECU.servo = newECU.servo + round(data.angular.z*0.01,5)      # input steering angle
    pubname.publish(newECU)

# state estimation node
def TwistToECU():

    # initialize node
    rospy.init_node('TwistToECU', anonymous=True)
    
    global pubname , newECU
    newECU = ECU() 
    # topic subscriptions / publications
    pubname = rospy.Publisher('ecu',ECU, queue_size = 100)

    rospy.Subscriber('turtle1/cmd_vel', Twist, callback_function)
    # set node rate
    loop_rate   = 50
    ts          = 1.0 / loop_rate
    rate        = rospy.Rate(loop_rate)
    t0          = time.time()

    rospy.spin()

if __name__ == '__main__':
    try:
       TwistToECU()
    except rospy.ROSInterruptException:
        pass
